src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +69,26 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            Assembly currentAssembly = typeof(Ldap).Assembly;
+
+            // Register callback that tries to load other libraries when the default library "libldap-2.5.so.0" not found
+            AssemblyLoadContext.GetLoadContext(currentAssembly).ResolvingUnmanagedDll += (assembly, ldapName) =>
+            {
+                if (assembly != currentAssembly || ldapName != Libraries.OpenLdap)
+                {
+                    return IntPtr.Zero;
+                }
+
+                // Try loading previous (libldap-2.4.so.2) and next (libldap-2.6.so.0) versions
+                if (NativeLibrary.TryLoad("libldap-2.4.so.2", out IntPtr handle) ||
+                    NativeLibrary.TryLoad("libldap-2.6.so.0", out handle))
```

Comment:
```suggestion
                if (NativeLibrary.TryLoad("libldap-2.4.so.2", out IntPtr handle) ||
                    NativeLibrary.TryLoad("libldap-2.6.so.0", out handle))
```

Is this the priority order we want? I would assume we want to try the latest first. Unsure the customer preference here, but I would think it is leaning toward the latest.

Reply:
2.6 is in preview, so I assume most likely it not included in any Linux distributions yet, but older versions, or even the latest versions of some distributions might still using the 2.4, therefore preferred checking 2.4 first.

Reply:
Instead of using a custom handler, would using a DllImportResolver be a better option?

Reply:
That was discussed and rejected at https://github.com/dotnet/runtime/pull/88851#discussion_r1262908436.

Reply:
> therefore preferred checking 2.4 first

The issue with this is users will then be required to either (a) remove 2.4 from their system or (b) update this BCL code path to try the preview version. I don't know how common having 2.4 and 2.5 on the same machine is, but if it is common then it makes trying out 2.6 difficult. I think we want 2.6 to be higher priority than 2.4.